src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/repository/VisitCustomRepository.kt

```diff
@@ -0,0 +1,45 @@
+package uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.repository
+
+import org.springframework.stereotype.Repository
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.data.VisitRoomCountResponse
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.data.filter.VisitFilter
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.AgencyInternalLocation
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.Visit
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.specification.VisitSpecification
+import javax.persistence.EntityManager
+import javax.persistence.PersistenceContext
+import javax.persistence.criteria.CriteriaBuilder
+import javax.persistence.criteria.CriteriaQuery
+import javax.persistence.criteria.Root
+
+interface VisitCustomRepository {
+  fun findRoomUsageCountWithFilter(filter: VisitFilter): List<VisitRoomCountResponse>
+}
+
+@Repository
+class VisitCustomRepositoryImpl : VisitCustomRepository {
+  @PersistenceContext
+  private val entityManager: EntityManager? = null
```

Comment:
this looks a bit naff - could it not be injected in the constructor?  Failing that would be better to make it a `lateinit var` and then wouldn't be nullable

Reply:
agreed